<?xml version="1.0"?>
<launch>

  <node pkg="relative_navigator" type="reference_trajectory_handler_node" name="reference_trajectory_handler_node" output="screen">
    <rosparam file="$(find relative_navigator)/config/param/reference_trajectory_handler.yaml" command="load"/>
  </node>

  <node pkg="relative_navigator" type="rel_pose_label_estimator_node.py" name="rel_pose_label_estimator_node" output="screen">
      <param name="observed_image_topic_name" value="/grasscam/image_raw/compressed"/>
      <!-- <param name="weight_path" value="/home/amsl/catkin_ws/src/vision-based_navigation/dnn_models/abstrelposnet/weights/ikuta_whole_area_12_03_with_dropout/best_acc.pt"/> -->
      <param name="direction_net_weight_path" value="/home/amsl/catkin_ws/src/vision-based_navigation/dnn_models/directionnet/weights/ikuta_whole_diff_cattiming/best_acc.pt"/>
      <param name="orientation_net_weight_path" value="/home/amsl/catkin_ws/src/vision-based_navigation/dnn_models/orientationnet/weights/ikuta_whole_1215/best_acc.pt"/>
  </node>

  <node pkg="relative_navigator" type="local_path_planner_node" name="local_path_planner_node" output="screen">
    <rosparam file="$(find relative_navigator)/config/param/local_path_planner.yaml" command="load"/>
  </node>

  <node pkg="relative_navigator" type="local_goal_generator_node" name="local_goal_generator_node" output="screen">
    <rosparam file="$(find relative_navigator)/config/param/local_goal_generator.yaml" command="load"/>
  </node>

  <node pkg="rviz" name="rviz_map_frame" type="rviz" args="-d $(find relative_navigator)/config/rviz/relative_navigator_map.rviz" output="screen"/>

  <node pkg="rviz" name="rviz_base_link_frame" type="rviz" args="-d $(find relative_navigator)/config/rviz/relative_navigator_base_link.rviz" output="screen"/>

  <node name="reference_image_republish" pkg="image_transport" type="republish" args="compressed raw">
      <remap from="in" to="/reference_image/image_raw" />
      <remap from="out" to="/reference_image/image_raw/" />
  </node>
  <node name="observe_image_republish" pkg="image_transport" type="republish" args="compressed raw">
      <remap from="in" to="/grasscam/image_raw/" />
      <remap from="out" to="/grasscam/image_raw/" />
  </node>

</launch>

